api/types.go

```diff
@@ -119,3 +119,15 @@ func NewListResponse[T, M any](items []M, fn func(item M) T) *ListResponse[T] {
 
 	return result
 }
+
+// PEM is a base64 encoded string, commonly used to store certificates and
+// private keys. PEM values will be normalized to remove any leading whitespace
+// and all but a single trailing newline.
+type PEM string
+
+// TODO: tests
+func (p *PEM) UnmarshalText(raw []byte) error {
+	v := string(raw)
+	*p = PEM(strings.TrimSpace(v) + "\n")
+	return nil
+}
```

Comment:
We could remove this for now, and introduce it later. It's not required to fix the problem we have now. 

If we did that I think the only test we'd be missing is one for the create destination endpoint to show that it doesn't trim whitespace for the `CA` field.